src/main/java/seedu/address/logic/parser/DeletePersonCommandParser.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.ParserUtil.arePrefixesPresent;
+
+import seedu.address.logic.commands.DeletePersonCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Name;
+
+/**
+ * Parses input arguments and creates a new {@code DeletePersonCommand} object
+ */
+public class DeletePersonCommandParser implements Parser<DeletePersonCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the
+     * {@code DeletePersonCommand} and returns a {@code DeletePersonCommand} object for
+     * execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeletePersonCommand parse(String args) throws ParseException {
+        try {
+            ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME);
+            if (!arePrefixesPresent(argMultimap, PREFIX_NAME) || !argMultimap.getPreamble().isEmpty()) {
```

Comment:
```suggestion
            boolean isPrefixNameAbsent = !arePrefixesPresent(argMultimap, PREFIX_NAME);
            boolean isPreamblePresent = !argMultimap.getPreamble().isEmpty();
            if (isPrefixNameAbsent || isPreamblePresent) {
```

Might this help reduce the complexity of the boolean expression?